spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/hazelcast/HazelcastServerConfiguration.java

```diff
@@ -93,14 +98,43 @@ private static Config loadConfig(URL configUrl) throws IOException {
 	@Configuration(proxyBeanMethods = false)
 	@ConditionalOnSingleCandidate(Config.class)
 	static class HazelcastServerConfigConfiguration {
-
 		@Bean
 		HazelcastInstance hazelcastInstance(Config config) {
 			return getHazelcastInstance(config);
 		}
 
 	}
 
+	private interface ConfigurationCustomizer {
```

Comment:
Why `private`? We don't want to expose customizers functionality to users or it's just the first phase?

Reply:
Until we have a concrete need for other customisers, I'd prefer to minimise the surface area of the public API.